### DISCUSSION TITLE
guides/custom-armor-trims/

### DISCUSSION THREAD

**Discussion by giscus** [misode/misode.github.io#325] [Site]
# guides/custom-armor-trims/

https://misode.github.io/guides/custom-armor-trims/

**Comment by JagerMeistars** [comment]
Thanks Misode!

**Comment by Plagiatus** [comment]
Damn, you're fast!

**Comment by LennyTheSniper** [comment]
"Since we haven't added the override for all the other armor types, they will default to the previous trim_type, in our case 0.8, which is diamonds."

I have a question: When making the models for every permutation of armor and the custom trim materials, when is it supposed to appear in game? I feel like I did every right step to add all 10 of my custom materials to only the iron chestplate, but it doesn't change the way the items looks in game. Does it appear when you do that step for every single permutation for the entire set of armor, or for ALL of the armors. I'm very worried that if I do a single step wrong, I have to search everywhere for where that mistake is.

**Reply by LennyTheSniper** [reply]
Also, a way to generate these using something like a python script would be nice to have. This is a pain to make

**Reply by peanutsponge** [reply]
I've made a python script to do just that: https://github.com/peanutsponge/Smithing/tree/Cosmetic_Only

**Comment by ijgeoffrey** [comment]
This is so useful, thank you! Quick question: "The trim pattern used will have no effect on the texture, only the trim material can be used in the model overrides."
Is this a limitation of the mechanic as written in the snapshot, or just something about the way you wrote this pack? I have a datapack idea that would utilize different trims to create different weapon models, but I can't do that as it stands?

**Reply by misode** [reply]
It's a limitation of the game. The trim pattern is not exposed to the model predicates.

**Reply by coucoul38** [reply]
maybe you could create custom crafting recipes that give an item with a different customModelData depending on what trim you crafted it with

**Comment by Ryhon0** [comment]
Hey, you forgot to mention you need to create a `trim_templates.json` file when creating templates, took me so long to realize why it's not working

**Reply by misode** [reply]
~~I don't think that file exists. Where is this documented?~~

Ah it's an item tag. When I wrote this guide it seemed to work fine without adding the item to the tag, maybe something changed?

**Comment by Trebuchet-Vault-Base** [comment]
What part of the 1x8 texture maps to what armor type?

**Comment by hivye** [comment]
Is it possible (now with the 1.20 update being released) to have each trim pattern apply a different texture? If so, how? I would like to add a custom texture for each trim.

**Reply by Darukshock** [reply]
Yes ! Inside `resourcepack/assets/minecraft/models/item/\[armor piece]_[material]_trim.json` , you can put any model you want !
Here's the vanilla model for assets/minecraft/models/item/chainmail_chestplate_quartz_trim.json`.
```{
  "parent": "minecraft:item/generated",
  "textures": {
    "layer0": "minecraft:item/chainmail_chestplate",
    "layer1": "minecraft:trims/items/chestplate_trim_quartz"
  }
}```
You can replace it with anything

**Comment by UltraHispeed-Elite** [comment]
so lets say I wanted to make a custom item using nbt data and use it as the trim for a custom item, does smithing tables allow the use of nbt modified items as trims

**Reply by Darukshock** [reply]
Infortunately, datapacks can't (yet) check for nbt during recipes. The alternatives would be to tell the player to drop items together/use a dropper/dispenser

**Comment by tobie-d** [comment]
can you just like change the texture of an armour trim in the files or something and make it a resource pack?

**Reply by Darukshock** [reply]
Yes, you can replace assets/minecraft/textures/item/`\<name>_armor_trim_smithing_template.png` with a resource pack.

**Comment by ghost** [comment]
Me and my friends are experimenting with making custom armor trims, and for one of them we wanted to have the default armor model be replaced with a custom one that we made, so that it would fit better with a particular trim.

And I don't know if there's a better way to phrase or ask this:
But Is it possible to have a custom armor model from a texture pack (netherite, diamond, gold, iron, chainmail, & or leather) overwrite the vanilla armor model, but only when a trim is applied to said-custom armor model. That way, you can have both the vanilla in-game armor textures & the custom armor model exist simultaneously without either one overwriting the other?

**Reply by Jaybug58** [reply]
Yes! If a pixel isn't part of the palate, then it will retain its original color. So if you want a trim to have a separate armor model, just include that as part of the template. Just do be aware that some armors cover more pixels than others (chainmail and netherite being the 2 that come to mind) and will remain uncovered.
